contrib/platform/src/com/sun/jna/platform/unix/X11.java

```diff
@@ -797,6 +797,8 @@ void XSetWMProperties(Display display, Window window, String window_name,
                           Pointer class_hints);
 
     int XSetWMProtocols(Display display, Window window, Atom[] atom, int count);
+    int XGetWMProtocols(Display display, Window w, Atom[] protocols_return, IntByReference count_return);
```

Comment:
Good catch - I didn't realise the array was allocated on the callee side.  If I want to add a helper method, would  `X.java` be a good place to put it?